modules/utxo-ord/src/inscriptions.ts

```diff
@@ -57,8 +66,84 @@ export function createOutputScriptForInscription(
     output: compiledScript,
     depth: 0,
   };
-  const payment = payments.p2tr({ redeems: [redeem] }, { eccLib });
+
+  // cal fees here
+  return payments.p2tr({ redeems: [redeem] }, { eccLib });
+}
+
+export type InscriptionRevealData = {
+  address: string;
+  revealSize: number;
+};
+
+/**
+ * @returns InscriptionRevealData
+ * @param createInscriptionParams & network
+ */
+export function createInscriptionRevealData(
+  createInscriptionParams: CreateInscriptionParams & { network: Network }
+): InscriptionRevealData {
+  const payment = createPaymentForInscription(createInscriptionParams);
+
+  const { output: commitOutput } = payment;
+  assert(commitOutput);
+  const commitAddress = address.fromOutputScript(commitOutput, createInscriptionParams.network);
+
+  // create fake dummy txn for size
+  const psbt = bitgo.createPsbtForNetwork({ network: createInscriptionParams.network as Network });
+  psbt.addInput({
+    hash: Buffer.alloc(32),
+    index: 0,
+    witnessUtxo: { script: commitOutput, value: BigInt(10239812038) },
+  });
+  psbt.addOutput({ script: commitOutput, value: BigInt(10000) });
+  // Fails with Error: Need tapBip32Derivation to sign Taproot with HD
+  // psbt.signTaprootInputHD(0, {
+  //   fingerprint: Buffer.alloc(64),
+  //   publicKey: Buffer.alloc(64),
+  //   derivePath(path: string): any {},
+  //   sign(hash: Buffer): Buffer {
+  //     return Buffer.alloc(64);
+  //   },
+  //   signSchnorr(hash: Buffer): Buffer {
+  //     return Buffer.alloc(64);
+  //   },
+  // });
+
+  // Fails withError: Input #0 has witnessUtxo but non-segwit script: 51204ae1d7b9cdd20c848b3ae7db4f32f4ec841cf05859e53650c64970384caa622a
+  psbt.signInput(0, {
+    publicKey: Buffer.alloc(64),
+    sign(hash: Buffer, lowR?: boolean): Buffer {
+      return Buffer.alloc(64);
+    },
+  });
+  const revealSize = psbt.extractTransaction().virtualSize();
```

Comment:
This is a bit over the top - pretty sure we can add a simple func that adds up the various input components and returns the vsize. This code here can be used for verification in tests.

Let's extend https://github.com/BitGo/BitGoJS/blob/8a0f0841eabd07478b6f40129e15e83954743fc9/modules/unspents/src/inputWeights.ts instead

Reply:
We're gonna need (essentially) this PSBT built to spend the thing any way, so why not use the _real_ size rather than the error prone attempt to add up the pieces?